### DISCUSSION TITLE
How to split ttc to ttf

### DISCUSSION THREAD

**Discussion by gachiemchiep** [fonttools/fonttools#2647] [Q&A] [answered]
Hello forks. 
How to use this tool to convert ttc into ttf ? 
There's an online tool at [https://transfonter.org/ttc-unpack](https://transfonter.org/ttc-unpack) that can split ttc into multiple ttf.

**Comment by anthrotype** [comment, marked as answer]
you can use `ttx` command line tool with the `-y` option for each sub-font in the ttc file to extract .ttx (XML) files for each of the sub-fonts, then again the `ttx` tool to recompile the .ttx files to individual binary font files.
Or you could use a script like this one:
 
```python
from fontTools.ttLib.ttCollection import TTCollection
import os
import sys

filename = sys.argv[1]
ttc = TTCollection(filename)
basename = os.path.basename(filename)
for i, font in enumerate(ttc):
    font.save(f"{basename}#{i}.ttf")
```

next time please use https://github.com/fonttools/fonttools/discussions to ask questions instead of filing an issue.

**Reply by yozhic** [reply]
Thanks @[anthrotype](https://github.com/anthrotype)! Just one small typo: the `sys` module has no attribute `args`, but `argv`. So, this worked for me:

```python
#!/usr/bin/env python

from fontTools.ttLib.ttCollection import TTCollection
import os
import sys

filename = sys.argv[1]
ttc = TTCollection(filename)
basename = os.path.basename(filename)
for i, font in enumerate(ttc):
    font.save(f"{basename}#{i}.ttf")
```
